src/chain.rs

```diff
@@ -0,0 +1,56 @@
+//! Chain specific utils
+
+use std::{collections::HashMap, fmt::Debug};
+
+use alloy_primitives::U256;
+use alloy_rpc_types::{Header, Transaction};
+use revm::primitives::{BlockEnv, SpecId, TxEnv};
+
+use crate::{
+    mv_memory::{LazyAddresses, MvMemory},
+    BuildIdentityHasher, MemoryLocation, TxIdx,
+};
+
+/// Custom behaviours for different chains & networks
+pub trait PevmChain: Debug + Clone + PartialEq {
+    /// The error type for [Self::get_block_spec].
+    type GetBlockSpecError: Debug + Clone + PartialEq;
+
+    /// The error type for [Self::get_gas_price].
+    type GetGasPriceError: Debug + Clone + PartialEq;
+
+    /// Get chain id.
+    fn id(&self) -> u64;
+
+    /// Get block's [SpecId]
+    fn get_block_spec(&self, header: &Header) -> Result<SpecId, Self::GetBlockSpecError>;
+
+    /// Get tx gas price.
+    fn get_gas_price(&self, tx: &Transaction) -> Result<U256, Self::GetGasPriceError>;
+
+    /// Build [MvMemory]
+    fn build_mv_memory(
+        &self,
+        hasher: &ahash::RandomState,
+        block_env: &BlockEnv,
+        txs: &[TxEnv],
+    ) -> MvMemory {
+        let block_size = txs.len();
```

Comment:
This should be the Ethereum implementation. The default implementation should just return `MvMemory::new(txs.len(), HashMap::with_hasher(BuildIdentityHasher::default()), LazyAddresses::default())`.

Reply:
Which would be cleaner defined as a `Default` implementation in `mv_memory`, for us to just spam `MvMemory:Default()` here.

Reply:
Wait we still need to pass `block_env` in. We can make two constructors or just write the long line here.

Reply:
Note that we cannot `impl Default for MvMemory` because `txs.len()` is required to initialize `last_locations`.